File: app/src/main/java/ru/alexgiltd/musicdb/presentation/artist/ArtistDetailsView.java
package ru.alexgiltd.musicdb.presentation.artist;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import java.util.List;

import ru.alexgiltd.musicdb.model.SimpleArtistModel;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface ArtistDetailsView extends MvpView {

    void showArtistName(String artistName);

    void showArtistImage(String artistImage);

    void showArtistDescription(String artistDesc);

    void showSimilarArtists(List<SimpleArtistModel> artistList);

    @StateStrategyType(OneExecutionStateStrategy.class)
    void showFavoriteArtistNotification(String artistName);

    @StateStrategyType(OneExecutionStateStrategy.class)
    void showError(String message);

    void onStartLoading();

    void onFinishLoading();

    @StateStrategyType(SkipStrategy.class)
    void openArtistDetails(String artistName);
}

File: app/src/main/java/ru/alexgiltd/musicdb/presentation/artist/ArtistDetailsView.kt
package ru.alexgiltd.musicdb.presentation.artist

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

import ru.alexgiltd.musicdb.model.SimpleArtistModel

@StateStrategyType(AddToEndSingleStrategy::class)
interface ArtistDetailsView : MvpView {

    fun showArtistName(artistName: String)

    fun showArtistImage(artistImage: String)

    fun showArtistDescription(artistDesc: String)

    fun showSimilarArtists(artistList: List<SimpleArtistModel>)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun showFavoriteArtistNotification(artistName: String)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun showError(message: String)

    fun onStartLoading()

    fun onFinishLoading()

    @StateStrategyType(SkipStrategy::class)
    fun openArtistDetails(artistName: String)
}
